odra-casper/livenet/src/casper_client.rs

```diff
@@ -0,0 +1,445 @@
+            .as_str()
+            .unwrap();
+        let bytes = hex::decode(result).unwrap();
+        let (value, _) = FromBytes::from_bytes(&bytes).unwrap();
+        Some(value)
+    }
+
+    fn wait_for_deploy_hash(&self, deploy_hash: DeployHash) {
+        let deploy_hash_str = deploy_hash.inner().to_string();
+        let time_diff = Duration::from_secs(15);
+        let final_result;
+
+        loop {
+            println!("Waiting {:?} for {:?}", &time_diff, &deploy_hash_str);
+            std::thread::sleep(time_diff);
+            let result: GetDeployResult = self.get_deploy(deploy_hash);
+            if !result.execution_results.is_empty() {
+                final_result = result;
+                break;
+            }
+        }
+
+        match &final_result.execution_results[0].result {
+            casper_types::ExecutionResult::Failure {
+                effect: _,
+                transfers: _,
+                cost: _,
+                error_message
+            } => {
+                panic!(
+                    "Deploy {:?} failed with error: {:?}",
+                    deploy_hash_str, error_message
+                );
+            }
+            casper_types::ExecutionResult::Success {
+                effect: _,
+                transfers: _,
+                cost: _
+            } => {
+                println!("Deploy {:?} successfully executed", deploy_hash_str);
+            }
+        }
+    }
+
+    fn new_deploy(&self, session: ExecutableDeployItem, gas: Balance) -> Deploy {
+        let timestamp = Timestamp::now();
+        let ttl = TimeDiff::from(10000000);
+        let gas_price = 1;
+        let dependencies = vec![];
+        let chain_name = String::from(self.chain_name());
+        let payment = ExecutableDeployItem::ModuleBytes {
+            module_bytes: Default::default(),
+            args: runtime_args! {
+                "amount" => gas
+            }
+        };
+
+        Deploy::new(
+            timestamp,
+            ttl,
+            gas_price,
+            dependencies,
+            chain_name,
+            payment,
+            session,
+            &self.secret_key,
+            Some(self.public_key())
+        )
+    }
+
+    fn post_request<T: DeserializeOwned>(&self, request: Value) -> T {
+        let client = reqwest::blocking::Client::new();
+        let response = client
+            .post(self.node_address_rpc())
+            .json(&request)
+            .send()
+            .unwrap();
+        let response: JsonRpc = response.json().unwrap();
+        let response = response.get_result().unwrap();
+        serde_json::from_value(response.clone()).unwrap()
+    }
+}
+
+fn find_wasm_file_path(wasm_file_name: &str) -> PathBuf {
+    let mut path = PathBuf::from("wasm").join(wasm_file_name);
+    let mut checked_paths = vec![];
+    for _ in 0..2 {
+        if path.exists() {
+            println!("Found wasm under {:?}", path);
```

Comment:
Maybe we should use logger like in cargo odra for such messages?

Reply:
done